hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/impl/NetworkUtilizationManagerImpl.java

```diff
@@ -0,0 +1,202 @@
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.hapi.node.base.SignatureMap;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.Transaction;
+import com.hedera.hapi.node.base.TransactionID;
+import com.hedera.hapi.node.state.congestion.CongestionLevelStarts;
+import com.hedera.hapi.node.state.throttles.ThrottleUsageSnapshot;
+import com.hedera.hapi.node.state.throttles.ThrottleUsageSnapshots;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.fees.congestion.MonoMultiplierSources;
+import com.hedera.node.app.hapi.utils.throttles.DeterministicThrottle;
+import com.hedera.node.app.service.mono.fees.congestion.MultiplierSources;
+import com.hedera.node.app.state.HederaState;
+import com.hedera.node.app.throttle.CongestionThrottleService;
+import com.hedera.node.app.throttle.HandleThrottleAccumulator;
+import com.hedera.node.app.throttle.NetworkUtilizationManager;
+import com.hedera.node.app.workflows.TransactionInfo;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.util.Arrays;
+import javax.inject.Inject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Implementation of {@link NetworkUtilizationManager}  that delegates to injected {@link HandleThrottleAccumulator} and {@link
+ * MultiplierSources}.
+ */
+public class NetworkUtilizationManagerImpl implements NetworkUtilizationManager {
+    private static final Logger log = LogManager.getLogger(NetworkUtilizationManagerImpl.class);
+    // Used to update network utilization after a user-submitted transaction fails the signature
+    // validity
+    // screen; the stand-in is a CryptoTransfer because it best reflects the work done charging fees
+    static final TransactionInfo STAND_IN_CRYPTO_TRANSFER = new TransactionInfo(
+            Transaction.DEFAULT,
+            TransactionBody.DEFAULT,
+            TransactionID.DEFAULT,
+            AccountID.DEFAULT,
+            SignatureMap.DEFAULT,
+            Bytes.EMPTY,
+            CRYPTO_TRANSFER);
+
+    private final HandleThrottleAccumulator handleThrottling;
+
+    private final MonoMultiplierSources multiplierSources;
+
+    @Inject
+    public NetworkUtilizationManagerImpl(
+            @NonNull final HandleThrottleAccumulator handleThrottling,
+            @NonNull final MonoMultiplierSources multiplierSources) {
+        this.handleThrottling = requireNonNull(handleThrottling, "handleThrottling must not be null");
+        this.multiplierSources = requireNonNull(multiplierSources, "multiplierSources must not be null");
+    }
+
+    @Override
+    public void trackTxn(@NonNull final TransactionInfo txnInfo, Instant consensusTime, HederaState state) {
+        track(txnInfo, consensusTime, state);
```

Comment:
Does it make sense to inline `track()`? `trackFeePayments()` can call this method directly.

Reply:
Updated.